### ISSUE TITLE
[question] The VAE is great with 8x32x32 compression but the construction results are bad?

### ISSUE THREAD

**Issue by eisneim** [Lightricks/LTX-Video#87]
just a question, is it normal that the reconstruction result is bad? so this means most of the hard work is don't by Dit?

## first frame of the input video
![截图 2024-12-31 19-30-50](https://github.com/user-attachments/assets/0b471ca1-e0e2-436e-ae30-fa35b84837b1)

## reconstructed video

https://github.com/user-attachments/assets/dc0d7af8-b6fb-42c1-8fc0-628176fdfa65

## using code inside Diffusers

```python
def prepare_latents(
    vae: AutoencoderKLLTXVideo,
    image_or_video: torch.Tensor,
    patch_size: int = 1,
    patch_size_t: int = 1,
    device: Optional[torch.device] = None,
    dtype: Optional[torch.dtype] = None,
    generator: Optional[torch.Generator] = None,
    precompute: bool = False,
) -> torch.Tensor:
    device = device or vae.device

    if image_or_video.ndim == 4:
        image_or_video = image_or_video.unsqueeze(2)
    assert image_or_video.ndim == 5, f"Expected 5D tensor, got {image_or_video.ndim}D tensor"

    image_or_video = image_or_video.to(device=device, dtype=vae.dtype)
    image_or_video = image_or_video.permute(0, 2, 1, 3, 4).contiguous()  # [B, C, F, H, W] -> [B, F, C, H, W]
    if not precompute:
        latents = vae.encode(image_or_video).latent_dist.sample(generator=generator)
        latents = latents.to(dtype=dtype)
        _, _, num_frames, height, width = latents.shape
        latents = _normalize_latents(latents, vae.latents_mean, vae.latents_std)
        latents = _pack_latents(latents, patch_size, patch_size_t)
        return {"latents": latents, "num_frames": num_frames, "height": height, "width": width}
    else:
        if vae.use_slicing and image_or_video.shape[0] > 1:
            encoded_slices = [vae._encode(x_slice) for x_slice in image_or_video.split(1)]
            h = torch.cat(encoded_slices)
        else:
            h = vae._encode(image_or_video)
        _, _, num_frames, height, width = h.shape

        # TODO(aryan): This is very stupid that we might possibly be storing the latents_mean and latents_std in every file
        # if precomputation is enabled. We should probably have a single file where re-usable properties like this are stored
        # so as to reduce the disk memory requirements of the precomputed files.
        return {
            "latents": h,
            "num_frames": num_frames,
            "height": height,
            "width": width,
            "latents_mean": vae.latents_mean,
            "latents_std": vae.latents_std,
        }

#---------------------------------------
use the latent produces by above code 
#---------------------------------------

from diffusers.utils import export_to_video
from diffusers.video_processor import VideoProcessor
with torch.no_grad():
    video =  vae.decode(lt, return_dict=False)[0]
pcc = VideoProcessor(vae_scale_factor=32)
vv = pcc.postprocess_video(video)[0]
export_to_video(vv, "data/test_rec2.mp4")

```
thanks for any reply !

**Comment by eisneim** [comment]
i get it, it's the problem of normalization

**Comment by Alienge** [comment]
@eisneim i met same question. How did you solve it in the end?   This issue #134

**Comment by eisneim** [comment]
@Alienge make user you normalize it after encode, and de-normlize it before decode; the new 0.9.5 VAE has much better quality

**Comment by Alienge** [comment]
I think normalize it after encode, and de-normalize it before decode is not influence the reconstruction the result. If I just want to check the reconstruction result, then this pair of inverse operations will have little effect on the reconstruction result. I also tried normalize it after encode, and de-normalize it before decode, which had little effect on the result. i will try new 0.9.5 VAE.

**Comment by Alienge** [comment]
Hi, i am get the result of new 0.9.5 VAE. It also has lots of artificial parts. Any advices? Thx

https://github.com/user-attachments/assets/f8e61c31-6ed8-4492-912f-0c68f59c6003

**Comment by eisneim** [comment]
@Alienge pls show your code, my test shows 0.9.5 official weights(not diffusers version) has much better reconstruction quality;
